internal/state/mailbox.go

```diff
@@ -153,53 +153,65 @@ func (m *Mailbox) GetMessagesWithoutFlagCount(flag string) int {
 }
 
 func (m *Mailbox) AppendRegular(ctx context.Context, literal []byte, flags imap.FlagSet, date time.Time) (imap.UID, error) {
-	internalIDString, err := rfc822.GetHeaderValue(literal, ids.InternalIDKey)
-	if err != nil {
-		return 0, err
-	}
-
-	if len(internalIDString) > 0 {
-		msgID, err := imap.InternalMessageIDFromString(internalIDString)
+	var appendIntoDrafts bool
+	// Force create message when appending to drafts so that IMAP clients can create new draft messages.
+	if !strings.EqualFold(m.mbox.Name, "Drafts") {
```

Comment:
can we base it on special use flag `\Drafts`, so connector can decide what name it's gonna be and controls the behaviour. I am not sure if we allowed localisation in Bridge yet.